apps/rush-lib/src/logic/buildCache/AzureStorageBuildCacheProvider.ts

```diff
@@ -197,15 +202,15 @@ export class AzureStorageBuildCacheProvider extends CloudBuildCacheProviderBase
       if (sasString) {
         const connectionString: string = this._getConnectionString(sasString);
         blobServiceClient = BlobServiceClient.fromConnectionString(connectionString);
-      } else if (!this.isCacheWriteAllowed) {
+      } else if (!this._isCacheWriteAllowedByConfiguration) {
```

Comment:
Is it possible to have a read-only cred? My guess is not, if we're assuming the existence of a cred implies write access. But if it does, then how do we differentiate? If we have a read-only cred and set "write allowed" to false, would we be able to use the cred to connect and read?

Reply:
You can, but something has to hydrate the cache, so I'd think the scenario we push users toward would be CI setting a write credential in an env variable.

If, for some reason, you want a read credential in CI, you can run the `rush update-cloud-credentials` command.